Source/WebKit/UIProcess/API/haiku/NavigationClient.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (C) 2019, 2024 Haiku, Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "NavigationClient.h"
+
+#include "APINavigation.h"
+#include "APIObject.h"
+#include "WebPageProxy.h"
+#include "WebView.h"
+#include "WebViewConstants.h"
+
+#include <Looper.h>
+#include <Message.h>
+#include <String.h>
+
+using namespace WebKit;
+
+void NavigationClient::didCommitNavigation(WebPageProxy& page, API::Navigation* navigation, API::Object* userData)
+{
+    BMessage message(DID_COMMIT_NAVIGATION);
```

Comment:
If all this is going to do is send BMessage to the page view, it doesn't need to be in the API directory. This directory should be reserved for the public API, that is, classes that the browser needs to directly interact with.

At least this is how it is handled by WebKitLegacy, but I think the idea is the same here?

Reply:
idk, the other ports seem to do similar things. Also, APINavigationClient.h is in the API folder.

I suppose the idea is to put things in the API folder if they form part of the API the client needs to know to use WebKit. Obviously, things used directly by the browser fall under this. But I would argue that the exact format and name of DID_COMMIT_NAVIGATION also counts as part of the API.